File: apps/desktop/src/types/store.ts
export interface Store {
  id: string;
  code: string;
  name: string;
  address: string | null;
  is_active: boolean;
  created_at: string;
  updated_at: string;
}

export interface CreateStoreInput {
  code: string;
  name: string;
  address?: string;
}

export interface UpdateStoreInput {
  id: string;
  name: string;
  address?: string;
}

export interface Device {
  id: string;
  store_id: string;
  device_name: string;
  is_active: boolean;
  registered_at: string;
  last_seen_at?: string | null;
}

export interface RegisterDeviceInput {
  store_id: string;
  device_name: string;
}

File: services/api/app/models/store.py
"""
Store ORM model — central PostgreSQL.

Minimal implementation for Issue 13 FK support (devices.store_id).
Full store management endpoints land in a later issue.
"""

from datetime import UTC, datetime

from sqlalchemy import Boolean, DateTime, String
from sqlalchemy.orm import Mapped, mapped_column

from app.db import Base


def _utc_now() -> datetime:
    return datetime.now(UTC)


class Store(Base):
    """Physical stock location — central record."""

    __tablename__ = "stores"

    id: Mapped[str] = mapped_column(String(36), primary_key=True)
    code: Mapped[str] = mapped_column(String(50), unique=True, nullable=False, index=True)
    name: Mapped[str] = mapped_column(String(255), nullable=False)
    address: Mapped[str | None] = mapped_column(String(500), nullable=True)
    is_active: Mapped[bool] = mapped_column(Boolean, default=True, nullable=False)
    created_at: Mapped[datetime] = mapped_column(
        DateTime(timezone=True), default=_utc_now, nullable=False
    )
    updated_at: Mapped[datetime] = mapped_column(
        DateTime(timezone=True), default=_utc_now, onupdate=_utc_now, nullable=False
    )
